runtime/scripts/jme-builtins.js

```diff
@@ -816,6 +816,7 @@ newBuiltin('sum',[sig.listof(sig.type('number'))],TNum,math.sum,{unwrapValues: t
 newBuiltin('sum',[TVector],TNum,math.sum);
 newBuiltin('prod',[sig.listof(sig.type('number'))],TNum,math.prod,{unwrapValues: true});
 newBuiltin('prod',[TVector],TNum,math.prod);
+newBuiltin('weighted_random',[[TNum,TNum]],TNum,math.weighted_random);
```

Comment:
The item can be any type, not just number. Use the `sig.` functions to define a nested signature. See the first definition of `prod` a couple of lines above for an example.

When the values are passed through to `math.weighted_random`, the pairs corresponding to each item are still JME list values, so the probability part needs to be unwrapped to a pure number.